libtenzir/builtins/functions/duration.cpp

```diff
@@ -107,6 +107,16 @@ class into_duration_plugin final : public function_plugin {
           [&](const null_type&) {
             check(b->AppendNulls(arg.length()));
           },
+          [&](const duration_type& t) {
+            diagnostic::warning("casting to `duration` has no effect")
```

Comment:
Nitpick: It's not really a cast. Maybe "parsing as …" instead? Or something else.